validation.go

```diff
@@ -0,0 +1,383 @@
+package pubsub
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"time"
+
+	peer "github.com/libp2p/go-libp2p-peer"
+)
+
+const (
+	defaultValidateConcurrency = 1024
+	defaultValidateThrottle    = 8192
+)
+
+// Validator is a function that validates a message.
+type Validator func(context.Context, peer.ID, *Message) bool
+
+// ValidatorOpt is an option for RegisterTopicValidator.
+type ValidatorOpt func(addVal *addValReq) error
+
+// validation represents the validator pipeline
```

Comment:
Some more colour on this doc would be nice. Suggestion: "the validator pipeline runs a sequence of user-configured validators per-topic. It is possible to adjust various concurrency parameters, such as the number of workers and the max number of simultaneous validations. The user can also attach inline validators that will be executed synchronously; this may be useful to prevent superfluous context-switching for lightweight tasks."

Reply:
ok, will do.

Reply:
done.